crates/core/src/token.rs

```diff
@@ -1032,6 +965,78 @@ pub mod testing {
         arb_established_address, arb_non_internal_address,
     };
 
+    impl std::ops::Add for Amount {
+        type Output = Self;
+
+        fn add(self, rhs: Self) -> Self::Output {
+            self.checked_add(rhs).unwrap()
+        }
+    }
+
+    impl std::ops::AddAssign for Amount {
+        fn add_assign(&mut self, rhs: Self) {
+            *self = self.checked_add(rhs).unwrap();
+        }
+    }
+
+    impl std::ops::Sub for Amount {
+        type Output = Self;
+
+        fn sub(self, rhs: Self) -> Self::Output {
+            self.checked_sub(rhs).unwrap()
+        }
+    }
+
+    impl std::ops::SubAssign for Amount {
+        fn sub_assign(&mut self, rhs: Self) {
+            *self = *self - rhs;
+        }
+    }
+
+    impl<T> std::ops::Mul<T> for Amount
+    where
+        T: Into<Self>,
+    {
+        type Output = Amount;
+
+        fn mul(self, rhs: T) -> Self::Output {
+            self.checked_mul(rhs.into()).unwrap()
+        }
+    }
+
+    impl std::ops::Mul<Amount> for u64 {
+        type Output = Amount;
+
+        fn mul(self, rhs: Amount) -> Self::Output {
+            rhs * self
```

Comment:
checked? And same for the ones below this

Reply:
as above